src/game.cpp

```diff
@@ -3366,6 +3371,10 @@ void Game::processPlayerInteraction(std::vector<aabb3f> &highlight_boxes,
 			playeritem = mlist->getItem(client->getPlayerItem());
 	}
 
+	if (playeritem.getDefinition(itemdef_manager).name.empty()) { // override the hand
+		if (InventoryList *hlist = local_inventory->getList("hand"))
```

Comment:
It used to be outside the if statement, but @nerzhul commented, I should put it inside the condition, see https://github.com/minetest/minetest/pull/2827#discussion-diff-53079522
But I agree it doesn't look clear. I can change it back if you want.